### PR TITLE
	ARMv7 SIMD: Fix clang compatibility part 2

### PR DESCRIPTION
**Pull request by mattsarett**
Sorry to do this to you.  I think there is a better fix to this issue.

It turns out that GCC does support UAL syntax (strbeq) if we provide the proper directive.

### CODE REVIEW

**Comment by dcommander** [comment]
Yeah, except that I already spent some time testing the previous fix to make sure that it didn't break anything.  We are in beta, so I am loathe to introduce fixes that might cause a regression, which is why I preferred the #ifdef approach.  With this new approach, I will need some assurances that the .syntax directive is supported in all ARM-compatible versions of GCC.  Can you provide such assurances or at least research which versions of GCC support that directive?

**Comment by rengolin** [comment]
The .syntax directive is supported in all versions of GCC and Clang since probably 2003, so you should be safe with all 3.x and 4.x GCCs as well as all 3.x Clangs.

While I cannot guarantee that adding ".syntax unified" to the assembly files will not bring additional syntax issues (ie. force you to change a few more pre-UAL instructions to the new style), it should not bring anything more than that, as UAL is *just* a new syntax.

In a nutshell, once it compiles, it should work in the same way.
